### PR TITLE
Added jsonschema for issue defined

### PR DESCRIPTION
**Pull request by BrandonRudiselNDSU**
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        "type",
+        "version",
+        "time"
+      ],
+      "additionalProperties": false
+    },
+    "data": {
+      "type": "object",
+      "properties": {
+        "issues": {
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: We want to get rid of the "issues" field, and pull out "type", "tracker", "id", and "uri" to the top level.

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      }
+    }
+  },
+  "required": [
+    "meta",
+    "data",
+    "links"
+  ],
+  "additionalProperties": false
+}
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: The braces end up unbalanced, this requires a few more closing braces.

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        "version",
+        "time"
+      ],
+      "additionalProperties": false
+    },
+    "data": {
+      "type": "object",
+      "properties": {
+        "issues": {
+          "type": "String",
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: Note that "String" is invalid, should be "string" for jsonschema

**Review by jaden-young — CHANGES_REQUESTED** [review]

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      },
+      "required": [
+        "id",
+        "type",
+        "version",
+        "time"
+      ],
+      "additionalProperties": false
+    },
+    "data": {
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: We also need a `customData` object as a member of `data`

File: schemas/EiffelIssueDefinedEvent/1.0.0.json (hunk 1 above)
Comment (jaden-young) [reply schemas/EiffelIssueDefinedEvent/1.0.0.json]: That is, if we delete everything here in `"data"`, we want to end up with something that looks like 
```json
    "data": {
      "type": "object",
      "properties": {
        "type": {
        
        },
        "tracker": {

        },
        "id": {

        },
        "uri": {

        },
        "title": {

        },
        "customData": {

        }
```
but with balanced braces and types filled in

**Review by jaden-young — CHANGES_REQUESTED** [review]

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+              "value": {
+              }
+            },
+            "required": [
+              "key",
+              "value"
+            ],
+            "additionalProperties": false
+          }
+        }
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: We want an `"additionalProperties": false` between this line and the next.

**Review by jaden-young — CHANGES_REQUESTED** [review]
Add `"additionalProperties: false"` to `data`

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        "type",
+        "version",
+        "time"
+      ],
+      "additionalProperties": false
+    },
+    "data": {
+      "type": "object",
+      "properties": {
+        "category": {
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: "category" should be "type"

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+              }
+            },
+            "required": [
+              "key",
+              "value"
+            ],
+            "additionalProperties": false
+          }
+        }
+      },
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: Before `"additionalProperties"`, we want a `"required"` block to require "type", "tracker", "id" and "uri"

File: schemas/EiffelIssueDefinedEvent/1.0.0.json (hunk 6 above)
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: `"category"` should be `"type"`

File: schemas/EiffelIssueDefinedEvent/1.0.0.json (hunk 7 above)
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json:143]: We want to add a `"required"` block here to require `"type"`, `"tracker"`, `"id"`, and `"uri"` to be present on the `"data"` object.

File: schemas/EiffelIssueDefinedEvent/1.0.0.json [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        },
+        "required": [
+          "type",
+          "target"
+        ],
+        "additionalProperties": false
+      }
+    }
+  }
+}
```
Comment (jaden-young) [inline schemas/EiffelIssueDefinedEvent/1.0.0.json]: Should add a newline at the end of the file

**Review by jaden-young — CHANGES_REQUESTED** [review]

File: schemas/EiffelIssueDefinedEvent/1.0.0.json (hunk 6 above)
Comment (BrandonRudiselNDSU) [reply schemas/EiffelIssueDefinedEvent/1.0.0.json]: That would end up as 
 "properties": {
        "type": {
          "type": "string",
          "enum": [
Are you sure?

Comment (jaden-young) [reply schemas/EiffelIssueDefinedEvent/1.0.0.json]: Yes. The first "type" is the name of the property, the second "type" is a key saying "the property named 'type' is a 'string'"

**Review by jaden-young — APPROVED** [review]

**Pull request merged**